static/js/ReaderApp.jsx

```diff
@@ -199,6 +200,13 @@ class ReaderApp extends Component {
     document.addEventListener('click', this.handleInAppClickWithModifiers, {capture: true});
     // Save all initial panels to recently viewed
     this.state.panels.map(this.saveLastPlace);
+    // Initialize entries for first-time visitors to determine if they are new or returning presently or in the future
+    if (Sefaria.isNewVisitor()) {
```

Comment:
I agree with this. The user might already be logged in and therefore a returning visitor, but you can also have a user that is a returning visitor who has never logged in.

Reply:
I think this specific part of the code is more easily understood if it's here than it would be if it was in sefaria.js, but maybe that is the better location.